src/api/users/src/routes/users.js

```diff
@@ -28,27 +28,41 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
-router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
+router.get('/', validateGetParams(), async (req, res, next) => {
+  // if req.query.page is 0 or unparsable set to 1, otherwise set to the req.query.page
+  const parseReqPage = parseInt(req.query.page, 10);
+  const page = !parseReqPage ? 1 : parseReqPage;
+  // if req.query.per_page is 0 or unparsable, set to 20, otherwise set to the min of either req.query.page or 20
+  const parseReqPerPage = Math.min(parseInt(req.query.per_page, 10), 20);
+  const perPage = !parseReqPerPage ? 20 : parseReqPerPage;
 
-    if (snapshot.empty) {
-      logger.debug(`All users were requested by ${req.ip} but could not be found.`);
-      res.status(404).json({
-        msg: `That collection could not be found, or contains no data.`,
-      });
-    } else {
-      logger.debug(`All users were requested by ${req.ip} and served successfully.`);
+  const users = [];
+  let userToStartAt;
 
-      const usersArray = [];
-      snapshot.forEach((doc) => {
-        usersArray.push(doc.data());
-      });
+  if (page === 1) {
```

Comment:
I'm unclear why you need to special-case 1 and 0, and further, why you're having them behave so differently, or why you'd not put the 0 case first in this list.

At the very least this needs a comment explaining your logic; however, I'm hesitant to have 3 separate code-paths through here like this for what should be doable with 1.  If we end up needing all 3, you need separate tests that exercise all 3 paths.

Reply:
This was actually needed before I started parsing and using conditions to ensure only clean values can make it in. 

`const userToStartAt = page === 1 ? 0 : perPage * (page - 1)` this currently works as expected. Would this be okay?
If the page specified is the first page, the value to get users would naturally start at user 0, otherwise start at `perPage * (page - 1)`

Reply:
Let me understand this properly with the current code with `perPage = 20`

If I start at page 1, I'm getting the first 20 users
If I start at page 0, I'm getting 21 - 40 users?

Reply:
@c3ho Requesting a page <= 1 always returns page 1 (for reasons that currently elude me, though I can always hardcode values <= 1 to always equal 1).

Requesting pages > 1 currently works as expected.

Reply:
Yes, the else case should def work cause 20 * (2 - 1) = 20,

But What I mean is do we need the (page === 0) in this case, isn't it the same as page === 1? Or am I missing something?

can we not do :

```
// Calculate how we're chunking the results
const startIndexAt = perPage * (page - 1)

// If page is <1 we start index at 0 otherwise we use the pagination
const userToStartAt = startIndexAt<= 0 ? 0 : startIndexAt
```

Reply:
The `page === 0` case is gone since the latest commit (🥳) and replaced by `const userToStartAt = page === 1 ? 0 : perPage * (page - 1)` which does exactly that. Currently with your conditional suggestion it works as expected when perPage or page are < 0 or 1